### PR TITLE
DOCS-1034,DOCS-1036: Add Viam Registry configuration

### PR DESCRIPTION
**Pull request by andf-viam**
Add docs for adding a module to a robot from the Viam Registry, with guidance on browsing modules, adding modules, configuring version updates, and configuring module-supplied modular resources.

## Staging

- [Configure a module on your robot](https://docs-test.viam.dev/80fc36a91abfd06f104cffe18a46b380fd89b402/public/extend/modular-resources/configure/)

### CODE REVIEW

**Review by Fahmina — APPROVED** [review]
LGTM!!!!

File: docs/extend/modular-resources/configure.md [hunk 1]
```diff
@@ -1,30 +1,171 @@
...
+- **Major (X.*)**: Only update to newer minor releases of the same major release branch.
+  The module will automatically restart and update itself whenever new updates within the same major release are available in the Viam Registry.
+  For example, use this option to permit a module with version `1.2.3` to update to version `1.2.4` or `1.3.0` but not `2.0.0` or `3.0.0`.
+- **Latest** Always update to the latest version of this module available from the Viam Registry as soon as a new version becomes available.
+
+When using the **Patch (X.Y.Z)** version type, you may select any patch version of the module from the **Version** drop down menu, including past versions if desired.
+
+The current deployed version of your module and the latest version of that module available from the Viam Registry are shown on this pane for your reference.
+
+{{% alert title="Updating production robots" color="caution" %}}
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:82]:
```suggestion
{{% alert title="Caution" color="caution" %}}
```
I *think* these are kinda supposed to be uniform--but will add to tomorrow's meeting agenda because I've been wondering

File: docs/extend/modular-resources/configure.md [hunk 2]
```diff
@@ -1,30 +1,171 @@
...
+For any version type other than **Patch (X.Y.Z)**, the module will upgrade as soon as an update that matches that specified version type is available, which will **restart the module**. If, for example, the module provides a motor component, and the motor is running, it will stop while the module updgrades.
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:83]:
```suggestion
For any version type other than **Patch (X.Y.Z)**, the module will upgrade as soon as an update that matches that specified version type is available, which will **restart the module**.
If, for example, the module provides a motor component, and the motor is running, it will stop while the module upgrades.
```
newline and upgrades typo

File: docs/extend/modular-resources/configure.md [hunk 3]
```diff
@@ -1,30 +1,171 @@
...
-Configure a module on your robot to make one or more modular resources available for configuration.
+Add and configure a module on your robot to make one or more modular resources available for configuration.
+You can [add a module from the Viam Registry](#add-a-module-from-the-viam-registry), or you can [code your own module and add it to your robot locally](#add-a-local-module-to-your-robot).
 
-If you are developing your own modular resource, you must first follow [these steps](/extend/modular-resources/create/) to code your own module and generate an executable.
-If you are using a pre-built modular resource, make sure you install the module and determine the filename of [the module's executable](/extend/modular-resources/create/#compile-the-module-into-an-executable).
+{{% alert title="Modules vs. modular resources" color="tip" %}}
 
-Follow these steps to configure the module and modular resource:
+A configured *module* can make one or more *modular resources* available for configuration.
```
[the remark this answers is not in the document]
Comment (JessamyT) [reply docs/extend/modular-resources/configure.md:16]: Also if this is still the case, and this tip stays here, it's confusing because it seems the user doesn't actually have to explicitly configure a module from the registry, but rather just a modular resource made available by that module, which implicitly autoconfigures the module??

File: docs/extend/modular-resources/configure.md [hunk 4]
```diff
@@ -1,30 +1,171 @@
...
+   {{<imgproc src="extend/modular-resources/configure/add-module-from-registry.png" resize="400x" declaredimensions=true alt="The add a component modal showing results for the intel realsense module ">}}
+
+1. When you have entered the name of the module that you would like to add to your robot, select the matching module in the search results and click the **Add module** button.
+
+When you add a module from the Viam Registry, the custom modular component it provides appears under the **Components** subtab like any other component.
+You can also [configure the module itself](#configure-a-module-from-the-viam-registry) from its configuration pane in the **Modules** subtab.
+
+{{<imgproc src="extend/modular-resources/configure/conf-component-from-module.png" resize="400x" declaredimensions=true alt="The components subtab of the config tab showing the camera component configuration pane for the realsense module">}}
+
+If the module requires that specific **Atrributes** be configured, click the **URL** link in the module's configuration pane to view the specific documentation for that module on the module's Github page.
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:45]:
```suggestion
If the module requires you to configure specific **Atrributes**, click the **URL** link in the module's configuration pane to view the documentation on that module's GitHub page.
```

File: docs/extend/modular-resources/configure.md [hunk 5]
```diff
@@ -1,30 +1,171 @@
...
+
+To delete a module added from the Viam Registry, click the trash can icon in the upper-right corner of the module configuration pane in the **Components** tab.
+
+### Configure a module from the Viam Registry
+
+Once you have added a module from the Viam Registry, you can view and configure the module from the **Modules** subtab:
+
+1. Navigate to the **Config** tab of your robot's page in [the Viam app](https://app.viam.com).
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:53]:
```suggestion
1. Navigate to the **Config** tab of your robot's page in the [Viam app](https://app.viam.com).
```
[opt nit] I've been doing it this way

File: docs/extend/modular-resources/configure.md [hunk 6]
```diff
@@ -1,30 +1,171 @@
...
+1. Click on the **Modules** subtab. All modules you have added to your robot appear under the **Deployed** section.
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:54]:
```suggestion
1. Click on the **Modules** subtab.
  All modules you have added to your robot appear under the **Deployed** section.
```

File: docs/extend/modular-resources/configure.md [hunk 7]
```diff
@@ -1,30 +1,171 @@
...
+
+This pane lists the models provided by the module, any [components](/components/) on your robot that are currently using those models, and allows you to configure [how the module updates](#configure-version-update-management-for-a-registry-module) when a new version is available from the Viam Registry.
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:56]: [opt] This is long. Consider an unordered list, though maybe that'd be awk.

File: docs/extend/modular-resources/configure.md [hunk 8]
```diff
@@ -1,30 +1,171 @@
...
+
+{{<imgproc src="extend/modular-resources/configure/conf-module-from-registry.png" resize="1000x" declaredimensions=true alt="The module subtab of the config tab showing the realsense custom module configuration pane includes the update management section showing version update management options version type, set to Patch (X.Y.Z) and version set to 0.0.3">}}
+
+#### Configure version update management for a Registry module
+
+When you add and configure a module on your robot, the module is automatically set to pin to the specific patch release version you added.
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:62]: [opt] "automatically set to pin to" sounds potentially hard to parse for some users but I don't have a better idea that's still succinct. But also, isn't "add and configure" redundant?
```suggestion
When you configure a module on your robot, by default it will pin to the specific patch release version you added.
```

File: docs/extend/modular-resources/configure.md [hunk 9]
```diff
@@ -1,30 +1,171 @@
...
+
+Depending on how you use the module, and what role your robot plays, you may wish to control how that deployed module updates when a new version of the module is made available in the Viam Registry.
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:64]: [opt]
```suggestion
If you wish to control how that deployed module updates when a new version of the module becomes available in the Viam Registry, you can change this setting.
```

File: docs/extend/modular-resources/configure.md [hunk 10]
```diff
@@ -1,30 +1,171 @@
...
+Updating to a newer version of a module brings new functionality and bug fixes, but requires restarting the module to apply the update.
+
+You can use the **Version type** menu of a module's configuration pane under the **Modules** subtab to configure version update behavior for your module:
 
+- **Patch (X.Y.Z)**: Do not update to any other version. This is the default.
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:69]:
```suggestion
- **Patch (X.Y.Z)**: Do not update to any other version.
  This is the default.
```

File: docs/extend/modular-resources/configure.md [hunk 11]
```diff
@@ -1,30 +1,171 @@
...
+- **Minor (X.Y.*)**: Only update to newer patch releases of the same minor release branch.
+  The module will automatically restart and update itself whenever new updates within the same minor release are available in the Viam Registry.
+  For example, use this option to permit a module with version `1.2.3` to update to version `1.2.4` or `1.2.5` but not `1.3.0` or `2.0.0`.
+- **Major (X.*)**: Only update to newer minor releases of the same major release branch.
+  The module will automatically restart and update itself whenever new updates within the same major release are available in the Viam Registry.
+  For example, use this option to permit a module with version `1.2.3` to update to version `1.2.4` or `1.3.0` but not `2.0.0` or `3.0.0`.
+- **Latest** Always update to the latest version of this module available from the Viam Registry as soon as a new version becomes available.
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:76]:
```suggestion
- **Latest**: Always update to the latest version of this module available from the Viam Registry as soon as a new version becomes available.
```

File: docs/extend/modular-resources/configure.md [hunk 12]
```diff
@@ -1,30 +1,171 @@
...
+
+The current deployed version of your module and the latest version of that module available from the Viam Registry are shown on this pane for your reference.
+
+{{% alert title="Updating production robots" color="caution" %}}
+For any version type other than **Patch (X.Y.Z)**, the module will upgrade as soon as an update that matches that specified version type is available, which will **restart the module**. If, for example, the module provides a motor component, and the motor is running, it will stop while the module updgrades.
 {{% /alert %}}
 
+### Configure a modular resource from a Registry module
+
+Once you have configured a module from the Viam Registry, you can add any number of the resources that module makes available to your robot by adding new components or services configured with your modular resources' new subtype or [model](/extend/modular-resources/key-concepts/#models).
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:88]:
```suggestion
Once you have configured a module from the Viam Registry, you can add any number of the resources that the module makes available to your robot by adding new components or services configured with your modular resources' [model](/extend/modular-resources/key-concepts/#models).
```
I *think* you can just say "model" since you can't configure a new subtype without a new model that implements it.

I realize that it works without the "the" but I trip on it without another "that" or "the" in there, esp. since such a long sentence.

File: docs/extend/modular-resources/configure.md [hunk 13]
```diff
@@ -1,30 +1,171 @@
...
+| `type` | string | **Required** | The {{< glossary_tooltip term_id="subtype" text="subtype">}} of the [API](/extend/modular-resources/key-concepts/#apis) (the third part of the {{< glossary_tooltip term_id="api-namespace-triplet" text="API namespace triplet">}}). |
+| `name` | string | **Required** | What you want to name this instance of your modular resource. |
+| `model` | string | **Required** | The full {{< glossary_tooltip term_id="model-namespace-triplet" text="model namespace triplet">}} of the modular resource's [model](/extend/modular-resources/key-concepts/#models). |
+| `depends_on` | array | Optional | The `name` of components you want to confirm are available on your robot alongside your modular resource. Often a [board](/components/board/). Unnecessary if you coded [implicit dependencies](/extend/modular-resources/key-concepts/#dependency-management). |
+
+All standard properties for configuration, such as `attributes` and `depends_on`, are also supported for modular resources.
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:100]: You have `depends_on` in the table above so this doesn't quite make sense. Maybe delete this sentence and add `attributes` to the table above?

File: docs/extend/modular-resources/configure.md [hunk 14]
```diff
@@ -1,30 +1,171 @@
...
+}
+```
+
+{{% /tab %}}
+{{% tab name="JSON Example" %}}
+
+The following is an example configuration for a base modular resource implementation.
+The configuration adds `acme:demo:mybase` as a modular resource from the module `my_base`.
+The custom model is configured as a component with the name "my-custom-base-1".
+You can send commands to the base according to the Viam [base API](/components/base/#api):
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:127]:
```suggestion
You can send commands to the base according to the Viam [base API](/components/base/#api).
```
I don't think the colon belongs, since what follows is the config, not the API info.

File: docs/extend/modular-resources/configure.md [hunk 15]
```diff
@@ -1,30 +1,171 @@
...
+            "name": "main-board",
+            "model": "pi"
+        },
+        {
+        "type": "base",
+        "name": "my-custom-base-1",
+        "model": "acme:demo:mybase",
+        "namespace": "rdk",
+        "attributes": {},
+        "depends_on": [ "main-board" ]
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:143]:
```suggestion
            "type": "base",
            "name": "my-custom-base-1",
            "model": "acme:demo:mybase",
            "namespace": "rdk",
            "attributes": {},
            "depends_on": [ "main-board" ]
```

File: docs/extend/modular-resources/configure.md [hunk 16]
```diff
@@ -1,30 +1,171 @@
...
+    }
+  ],
+  "modules": [ ... ] // < INSERT YOUR MODULE CONFIGURATION >
+}
+```
+
+{{% /tab %}}
+{{% tab name="JSON Example" %}}
+
+The following is an example configuration for a base modular resource implementation.
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:124]:
```suggestion
The following is an example configuration for a base modular resource implementation.
It also includes the configuration for a built-in model of [board](/components/board/) on which this modular base model depends.
```
Maybe? Or something like that, to acknowledge the fact that the JSON contains more than just the modular resource.

File: docs/extend/modular-resources/configure.md [hunk 17]
```diff
@@ -1,30 +1,171 @@
...
+}
+```
+
+{{% /tab %}}
+{{% /tabs %}}
+
+## Add a local module to your robot
+
+If you are developing your own modular resource, and intend to deploy it to your robot locally, first follow [these steps](/extend/modular-resources/create/) to code your own module and generate an executable.
+If you are using a pre-built modular resource, make sure you install the module and determine the filename of [the module's executable](/extend/modular-resources/create/#compile-the-module-into-an-executable).
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:161]: Why would someone use a pre-built one locally? I wonder if it'd make more sense to introduce this use case in a separate paragraph here? Also what exactly is meant by pre-built? Is it something from the registry, but downloaded independently? I also fear that someone might confuse "pre-built" with "built-in" and get real confused :D

File: docs/extend/modular-resources/configure.md [hunk 18]
```diff
@@ -1,30 +1,171 @@
...
 weight: 30
 type: "docs"
 tags: ["server", "rdk", "extending viam", "modular resources", "components", "services"]
-description: "Use the Viam module system to implement custom resources that can be included in any Viam-powered robot."
+description: "Configure a module on your robot."
 no_list: true
 ---
 
-Configure a module on your robot to make one or more modular resources available for configuration.
+Add and configure a module on your robot to make one or more modular resources available for configuration.
```
Comment (JessamyT) [inline docs/extend/modular-resources/configure.md:11]:
```suggestion
Add a module and configure a modular resource to add extended functionality to your robot.
```
or perhaps
```suggestion
To add a modular resource to your robot, first configure a module and then configure the modular component or service.
```
or maybe delete this sentence entirely? It's a bit tautological I think.
Also isn't "add and configure" redundant? You have to configure the module in the `modules` section of your config, and configure the resource in the `components` or `services` part, right? But you could call both of those things "adding" or "configuring"

**Review by JessamyT — COMMENTED** [review]
Nice. Most comments are more about clarity/perhaps my own confusion than about anything desperately in need of modification.

File: docs/extend/modular-resources/configure.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (andf-viam) [reply docs/extend/modular-resources/configure.md:16]: Somewhat. Both module and resource-provided-by module are configurable, this page for both. Adding the module-provided resource does implicitly auto-add the module, but with default settings (like pin to patch). You may wish to also-configure module beyond resource-provided-by-module.

Inlined guidance (removed as-tip, now as-paragraph)

This copy also updated in [the other PR](https://github.com/viamrobotics/docs/pull/1705) with further clarifications.

File: docs/extend/modular-resources/configure.md [hunk 19]
```diff
@@ -1,30 +1,171 @@
...
+A configured *module* can make one or more *modular resources* available for configuration.
+See [Key Concepts of Modular Resource APIs](/extend/modular-resources/key-concepts/) for more information.
 
-1. [Save the executable](#make-sure-viam-server-can-access-your-executable) in a location your `viam-server` instance can access.
-2. [Add a **module**](#configure-your-module) referencing this executable to the configuration of your robot.
-3. [Add a new component or service](#configure-your-modular-resource) referencing the custom resource provided by the configured **module** to the configuration of your robot.
+{{% /alert %}}
 
-{{% alert title="Modules vs. modular resources" color="tip" %}}
+## Add a module from the Viam Registry
```
[the remark this answers is not in the document]
Comment (andf-viam) [reply docs/extend/modular-resources/configure.md:21]: I believe so, haven't heard either way. Awaiting Product-submitted DOCS ticket to counter, if so.

File: docs/extend/modular-resources/configure.md [hunk 20]
```diff
@@ -1,30 +1,171 @@
...
+   - Start typing to search for modules by name.
+     Modules available from the Viam Registry will be listed under the `From Registry` section of the search results.
+   - [Browse the Viam Registry](https://app.viam.com/modules) directly to search available modules.
+
+   {{<imgproc src="extend/modular-resources/configure/add-module-from-registry.png" resize="400x" declaredimensions=true alt="The add a component modal showing results for the intel realsense module ">}}
+
+1. When you have entered the name of the module that you would like to add to your robot, select the matching module in the search results and click the **Add module** button.
+
+When you add a module from the Viam Registry, the custom modular component it provides appears under the **Components** subtab like any other component.
+You can also [configure the module itself](#configure-a-module-from-the-viam-registry) from its configuration pane in the **Modules** subtab.
```
[the remark this answers is not in the document]
Comment (andf-viam) [reply docs/extend/modular-resources/configure.md:41]: Good callout, updated!

File: docs/extend/modular-resources/configure.md (hunk 5 above)
Comment (andf-viam) [reply docs/extend/modular-resources/configure.md:53]: 146 v 79, going with more common, at least for now.

File: docs/extend/modular-resources/configure.md (hunk 12 above)
Comment (andf-viam) [reply docs/extend/modular-resources/configure.md:88]: Have also tripped on same, updated. But also refactored in https://github.com/viamrobotics/docs/pull/1705

File: docs/extend/modular-resources/configure.md (hunk 13 above)
Comment (andf-viam) [reply docs/extend/modular-resources/configure.md:100]: Leaving this as-found for now.

File: docs/extend/modular-resources/configure.md (hunk 17 above)
Comment (andf-viam) [reply docs/extend/modular-resources/configure.md:161]: This is following shortly, separately.

File: docs/extend/modular-resources/configure.md (hunk 18 above)
Comment (andf-viam) [reply docs/extend/modular-resources/configure.md:11]: Addressed also in https://github.com/viamrobotics/docs/pull/1705.
